cetmix_tower_yaml/models/cx_tower_yaml_mixin.py

```diff
@@ -115,6 +128,19 @@ def _get_fields_for_yaml(self):
         """
         return ["reference"]
 
+    def _get_force_x2m_resolve_models(self):
+        """List of models that will always try to be resolved
+        when referenced in x2m related fields.
+
+        This is useful for models that should always to use existing records
+        instead of creating new ones when referenced in x2m related fields.
+        Such as variables or tags.
+
+        Returns:
+            List: list of models that will always try to be resolved
+        """
+        return ["cx.tower.variable", "cx.tower.tag", "cx.tower.os"]
```

Comment:
Thank you @GabbasovDinar , I understand your concerns.
 
> In its current implementation, the _get_force_x2m_resolve_models method defines a hardcoded list of models (cx.tower.variable, cx.tower.tag, cx.tower.os). While this might work for some specific use cases, it introduces the following issues:
> 
> Reduces Flexibility: By hardcoding model names into the abstract model, it becomes less generic. Any changes to the list require modifications to the base model, which may not be desirable in a modular and extensible system.

This is not the case because those model are generic. Which means they must be added by default, not as an option. Otherwise we need to inherit and redefine this method for every new model which is not the case.
> 
> Contradicts Abstract Model Design: Abstract models should be neutral and avoid embedding specific implementation details unless strictly necessary. This method introduces a dependency on specific models, which might not be relevant for all use cases or inheriting models.

If there is a local need to modify this list it can be done at the model level, like it's done [here.](https://github.com/cetmix/cetmix-tower/blob/59902b4086c6bc4de38c4801c97dbbb6b90053ff/cetmix_tower_yaml/models/cx_tower_server_template.py#L43)